### PR TITLE
Set events createdAt to the time domain event is sent

### PR DESCRIPTION
**Pull request by xaviapa**

### CODE REVIEW

File: tests/Jobs/SendDomainEventsTest.php [hunk 1]
```diff
@@ -35,94 +47,82 @@ public function messagesToSendProvider(): array
...
     /**
      * @test
      * @dataProvider messagesToSendProvider
      */
-    public function whenMessageCannotBeSendItShouldTryAgainLater($messages)
+    public function whenMessageCannotBeSendItShouldTryAgainLater(array $messages)
     {
-        $amqpMiddleware = Mockery::mock(AmqpMiddleware::class);
-        $warningMessage = "The event could't be sent. Retrying message: " . json_encode($messages);
+        $warningMessage = "The event couldn't be sent. Retrying message: " . json_encode($messages);
```
Comment (jeurrutia) [inline tests/Jobs/SendDomainEventsTest.php:109]: is json_enconde wrong indented?

File: tests/Console/Commands/EmitAllEventsTest.php [hunk 2]
```diff
@@ -30,7 +31,7 @@ public function whenRunCommandItShouldResendAllTheCurrentDomainEvents(): void
         $this->app->register('Softonic\TransactionalEventPublisher\ServiceProvider');
         $this->artisan('event-sourcing:emit-all')->run();
 
-        $this->assertCount(4, $this->dispatchedJobs);
+        self::assertCount(4, $this->dispatchedJobs);
```
Comment (jeurrutia) [inline tests/Console/Commands/EmitAllEventsTest.php:37]: It would be nice to avoid magic numbers

File: .travis.yml [hunk 3]
```diff
@@ -4,10 +4,6 @@ sudo: false
 
 matrix:
   include:
-    - php: 7.2
```
Comment (joskfg) [inline .travis.yml:7]: Why are not checking any php version?

We are just checking vs PHP 8

If travis is not working, we could try github actions.

File: README.md [hunk 4]
```diff
@@ -37,15 +37,24 @@ php artisan vendor:publish --provider="Softonic\TransactionalEventPublisher\Serv
...
 
 #### Sync AMQP middleware 
 
 To use the sync AMQP you just need to configure the AMQP connection using the configuration file or environmental variables. 
 As you could see, in the configuration you won't be able to define a queue. This is because the library just publishes the message to an exchange and is the events collector responsability to declare the needed queues with the needed bindings.
  
-#### Async AMQP middleware
+#### Async middleware
 
+You need to configure the middleware you want to publish the events. For example:
```
Comment (joskfg) [inline README.md]: Explain that actually the async middleware is just reposnsible to delay the action, but you need to configure a real middle that is who pusblish the messages

File: README.md [hunk 5]
```diff
@@ -37,15 +37,24 @@ php artisan vendor:publish --provider="Softonic\TransactionalEventPublisher\Serv
...
+```php
+    /*
+     |--------------------------------------------------------------------------
+     | Middleware that publishes the events when using AsyncMiddleware.
+     |--------------------------------------------------------------------------
+     */
+    'event_publisher_middleware' => \Softonic\TransactionalEventPublisher\EventStoreMiddlewares\AmqpMiddleware::class,
+```
 You need to do the Sync AMQP middleware steps and continue with these:
```
Comment (joskfg) [inline README.md]: You don't need to do this if you don't want to use the AMQPMiddleware as delayed middleware, you still need to do the steps bellow this line.

File: config/transactional-event-publisher.php [hunk 6]
```diff
@@ -13,7 +13,14 @@
     | Middleware class where the event messages will be stored.
     |--------------------------------------------------------------------------
     */
-    'middleware' => \Softonic\TransactionalEventPublisher\EventStoreMiddlewares\AmqpMiddleware::class,
+    'middleware' => \Softonic\TransactionalEventPublisher\EventStoreMiddlewares\AsyncMiddleware::class,
```
Comment (joskfg) [inline config/transactional-event-publisher.php]: Why are we changing the default middleware?

**Review by joskfg — CHANGES_REQUESTED** [review]

**Review by joskfg — APPROVED** [review]

**Pull request merged**